### DISCUSSION TITLE
Telegram Bot tidak menerima respon dari pythonanywhere ataupun gitpod

### DISCUSSION THREAD

**Discussion by intanmsari** [teamalgoritma/community#913] [Q&A] [answered]
hi, saat ini saya sedang mengalami kesulitan dimana telegram bot tidak muncul respons dari pythonanywhere ataupun gitpod.

Step:
1. deploy di github > https://github.com/intanmsari/telebot-main
2. git clone di pythonanywhere dan running pip install
3. configure directory path dan configure WSGI file
4. reload dan cek bot sdh running
![image](https://github.com/user-attachments/assets/ee65ce91-f23c-4a94-98ab-0619043cc15e)
5. cek di access log:
![image](https://github.com/user-attachments/assets/8e3adf4d-dc91-4b78-8627-cad030edbc62)
6. test di telegram bot > DACP_ZEUS_INTAN_BOT, tidak ada respon dari app.
![image](https://github.com/user-attachments/assets/9d425122-a52e-482f-b314-e6572db7ef32)

jika ada insights knp tdak muncul respons dari app nya ya padahal kalo di cek bot sdh running dan liat di access log juga ada pergerakan.

Notes: 
[intanphy.pythonanywhere.com.server.log.zip](https://github.com/user-attachments/files/16828616/intanphy.pythonanywhere.com.server.log.zip)

Terima kasih sebelumnya 🫰

**Comment by saskia-dwi-ulfah** [comment]
Dear Bu @intanmsari 

Terima kasih telah bertanya melalui Github community Algoritma.
Sebelumnya kami izin mengonfirmasi, Bu, apakah ibu sudah menambahkan `import sys` pada configuration file WSGI-nya?

**Reply by intanmsari** [reply]
![image](https://github.com/user-attachments/assets/85564d1a-3561-49f1-b2fd-cdb2485c71b1)

berikut SS nya kak. sudah seperti instruksinya

**Reply by saskia-dwi-ulfah** [reply]
Dear Bu @intanmsari 

Terima kasih atas konfirmasinya. Sepertinya memang ada bug internal dari kedua platform deploy ini bu, yang harus kita tracing, walaupun sudah mengikuti step-step seperti pada video pengerjaan capstone. Jika tertarik untuk eksplorasi tentang hal ini agar bot-nya bisa berfungsi, ibu bisa coba ulangi kembali pengerjaannya dari awal. Adapun jika ibu tidak ingin mengulang dari awal karena sudah mengikuti setiap step dengan benar, ibu bisa langsung mengumpulkan apa yang sudah dikerjakan.

Demikian yang dapat kami sampaikan. Semoga dapat membantu.

Best Regards,
Team Algoritma

**Comment by hwulanayu** [comment, marked as answer]
Dear Bu @intanmsari

Silakan coba menambahkan parameter `threaded=False` di inisiasi Telebot pada file bot_local.py. Contoh kode:
```
load_dotenv()
TOKEN = os.environ['TOKEN']
bot = telebot.TeleBot(TOKEN, threaded=False)
```

Jika sudah dideploy, tambahkan pada pythonanywhere di bagian File > klik directory yang sudah dibuat untuk capstone telebot > klik `bot_local.py` file  dan edit parameter sesuai kode di atas > klik Save > reload ulang di tab Web.

Best Regards,
Team Algoritma

**Comment by adizaen** [comment]
Dear @intanmsari 

Terima kasih telah bertanya melalui Github discussions Algoritma. Untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

**Best regards,**
Team Algoritma
